model/DataObjects.inc.php

```diff
@@ -648,6 +655,14 @@ public static function delete($libraryID, $key) {
 		self::unload($obj->id);
 		
 		if ($deleted) {
+			// If the attachment item was successfully deleted, check
+			// if other items are using this storageFileID,
+			// and if not, delete the row with storageFileID and libraryID from
+			// storageFileLibraries table
+			if ($type == 'item' && $obj->isImportedAttachment()) {
+				Zotero_Storage::deleteFileLibraryReference($storageFileID, $libraryID);
+			}
+			
```

Comment:
I'm not sure when `$deleted` can actually be false (i.e., when this function would be called without there being an `items` row to remove), but this can just go up in the section above.